### ISSUE TITLE
Toil trying to write to /var/run/user and failing

### ISSUE THREAD

**Issue by oneillkza** [ComparativeGenomicsToolkit/cactus#871]
I'm getting the following error:

```
FileNotFoundError: [Errno 2] No such file or directory: '/var/run/user/1639/toil/695e5e325d295c23888dc2b980357bf3/tmpxrtkm5yx.jobState'
```

This seems related to the below bug report for Toil:

https://pullanswer.com/questions/toil-uses-var-run-user-uid-if-it-sees-it-whether-or-not-it-is-actually-available-to-toils-current-session

And may be fixed in the upcoming 6.0.0 release:

https://github.com/DataBiosphere/toil/wiki/Draft-Changelog

> Only use /var/run/user if XDG tells us we have it in our session. Otherwise we will try other places, including /run/lock/toil. (#4170)

Meanwhile the solution may be to downgrade Toil to 5.6.0

**Comment by oneillkza** [comment]
Confirming that running the following within the virtual environment got past this error, as a temporary fix:

```
python3 -m pip install -U toil[aws]==5.6.0
```

Except, on my system, that then leads to various errors as a result of race condition bugs in Toil when it's running on an NFS file system...

**Comment by oneillkza** [comment]
So I tried setting --workDir to a local tmp disk space, and now I get the following very cryptic error off the bat:

```
[2022-12-21T14:11:43-0800] [MainThread] [I] [toil.leader] Issued job 'progressive_workflow' kind-progressive_workflow/instance-niihfy3e v5 with job batch system ID: 4 and cores: 1, disk: 2.0 Gi, and memory: 14.9 Gi
[2022-12-21T14:12:19-0800] [MainThread] [W] [toil.leader] Job failed with exit value 1: 'progressive_workflow' kind-progressive_workflow/instance-niihfy3e v5
Exit reason: None
[2022-12-21T14:12:19-0800] [MainThread] [W] [toil.leader] No log file is present, despite job failing: 'progressive_workflow' kind-progressive_workflow/instance-niihfy3e v5
[2022-12-21T14:12:19-0800] [MainThread] [W] [toil.job] Due to failure we are reducing the remaining try count of job 'progressive_workflow' kind-progressive_workflow/instance-niihfy3e v5 with ID kind-progressive_workflow/instance-niihfy3e to 1
```

**Comment by oneillkza** [comment]
May also be related to https://github.com/DataBiosphere/toil/issues/4218

**Comment by oneillkza** [comment]
OK, based on testing I did in https://github.com/DataBiosphere/toil/issues/4307, the /var/usr issue is solved with the latest Toil, and should be solved by updating to the impending 5.8 release.

The cryptic Slurm errors were my own mistake in managing tmp directories.

**Comment by oneillkza** [comment]
I'm closing this, although the fix still requires using dev versions of code. But the next release of cactus should include the necessary release of Toil (5.8).
